lib/write.js

```diff
@@ -1,60 +1,62 @@
+'use strict';
+
 var _ = require('underscore');
+var AdapterWrite = require('juttle/lib/runtime/adapter-write');
+var errors = require('juttle/lib/errors');
 var db = require('./db');
-var Juttle = require('juttle/lib/runtime').Juttle;
 var values = require('juttle/lib/runtime/values');
-
+var Promise = require('bluebird');
 var logger = require('juttle/lib/logger').getLogger('sql-db-write');
 
-var Write = Juttle.proc.sink.extend({
-    procName: 'write-sql',
-
-    allowed_options: ['table'],
+const ALLOWED_OPTIONS = ['table'];
 
-    initialize: function(options, params) {
+class WriteSql extends AdapterWrite {
+    constructor(options, params) {
+        super(options, params);
         this.handleOptions(options);
         this.inserts_in_progress = 0;
         this.eof_received = false;
-    },
+        this.writePromise = Promise.resolve();
+    }
 
-    handleOptions: function(options) {
+    handleOptions(options) {
         logger.debug('init options:', options);
-        var unknown = _.difference(_.keys(options), this.allowed_options);
+        var unknown = _.difference(_.keys(options), ALLOWED_OPTIONS);
```

Comment:
As of https://github.com/juttle/juttle/pull/317 you can just implement an `allowedOptions()` method that returns `['table']` and the base class will take care of this.